tests/src/Behat/OeThemeTestContext.php

```diff
@@ -337,44 +368,20 @@ public function setCorporateLibrary(string $component_library): void {
   }
 
   /**
-   * Assert presence or absence of footer blocks.
+   * Determines footer type.
    *
-   * @param string $component_library
-   *   Component library name: either 'European Commission' or 'European Union'.
-   * @param bool $presence
-   *   Wheres presence (TRUE) or absence (FALSE) should be asserted.
+   * The ECL doesn't provide a way to determine type of the footer except as
+   * to check the existence of the logo.
+   * 'European Union' has logo, 'European Commission' doesn't.
+   *
+   * @return string
+   *   'European Union' or 'European Commission'.
    */
-  public function assertFooter(string $component_library, bool $presence): void {
-    // Map component library with the expected logo count from markup.
-    // This is necessary as the ECL gives us no other ways of determining
-    // which footer is which.
-    $expected = [
-      'European Commission' => 0,
-      'European Union' => 1,
-    ];
-
-    // Make sure a corporate footer is present on the page.
-    $this->assertSession()->elementExists('css', 'footer.ecl-footer-' . $this->getEclBranding());
-
-    $page = $this->getSession()->getPage();
-    $logo = $page->findAll('css', 'footer.ecl-footer-' . $this->getEclBranding() . ' img.ecl-footer-' . $this->getEclBranding() . '__logo-image-desktop');
-
-    // Assert presence or absence of given footer block.
-    $logo_found = count($logo) === $expected[$component_library];
-
-    if ($logo_found !== $presence) {
-      $expectation = $presence ? 'present' : 'absent';
-      throw new \Exception("The {$component_library} corporate footer block was expected to be {$expectation} but it is not.");
-    }
-
-    if ($logo && $presence) {
-      $link = $page->find('css', '.ecl-footer-' . $this->getEclBranding() . '__logo-link');
-      Assert::assertEquals('https://european-union.europa.eu/index_en', $link->getAttribute('href'));
-      Assert::assertEquals(t('Home - European Union')->__toString(), $link->getAttribute('aria-label'));
-      $img = $link->find('css', 'img');
-      Assert::assertEquals(t('European Union flag')->__toString(), $img->getAttribute('alt'));
-      Assert::assertEquals(t('European Union')->__toString(), $img->getAttribute('title'));
```

Comment:
I don't think the whole refactoring of this method + 2 other methods should be done in this PR since the scope of the issue is to simply assert the string which you now pass in different scenarios (being the same for each language).

Reply:
There is a comment above 
```
assertFooter includes 3 functions - checks EU footer, checks EC footer, ensures that footer doesn't exist.
For logic simplification I've rewritten it, otherwise it is mess of conditions.
```